app/authors/CreateAuthorForm.js

```diff
@@ -0,0 +1,132 @@
+"use client";
+
+import { gql, useMutation } from "@apollo/client";
+import { useRouter } from "next/navigation";
+import { useEffect, useState, useTransition } from "react";
+
+const CREATE_AUTHOR = gql`
+  mutation CreateAuthor(
+    $name: String!
+    $biography: String!
+    $profilePhotoUrl: String
+    $date_of_birth: String!
+  ) {
+    addAuthor(
+      name: $name
+      biography: $biography
+      profilePhotoUrl: $profilePhotoUrl
+      date_of_birth: $date_of_birth
+    ) {
+      id
+      name
+      biography
+      profilePhotoUrl
+      date_of_birth
+    }
+  }
+`;
+
+export default function CreateAuthorForm() {
+  const [name, setName] = useState("");
+  const [biography, setBiography] = useState("");
+  const [profilePhotoUrl, setProfilePhotoUrl] = useState("");
+  const [date_of_birth, setDateOfBirth] = useState("");
```

Comment:
think of nodes with lot of fields, this is not very scalable if we add states for each fields, any better way to do this.